math/benches/msm.rs

```diff
@@ -0,0 +1,124 @@
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use lambdaworks_math::cyclic_group::IsCyclicBilinearGroup;
+use lambdaworks_math::elliptic_curve::traits::HasDistortionMap;
+use lambdaworks_math::msm::naive::Naive;
+use lambdaworks_math::{
+    elliptic_curve::{
+        curves::test_curve::{QuadraticNonResidue, ORDER_P},
+        element::EllipticCurveElement,
+        traits::HasEllipticCurveOperations,
+    },
+    field::{
+        element::FieldElement, fields::u64_prime_field::U64PrimeField,
+        quadratic_extension::QuadraticExtensionField,
+    },
+    msm::{pippenger, MSM},
+};
+use rand::rngs::StdRng;
+use rand::{Rng, SeedableRng};
+
+type FE = FieldElement<U64PrimeField<ORDER_P>>;
+
+macro_rules! bench_msm_for_length {
+    ( $func:ident, $a:expr ) => {
```

Comment:
This is a macro that generates a criterion benchmark for a specified MSM length.
The idea is that it generates a set of N random field and group elements, and:
- Benchmarks the MSM with the Naive implementation.
- Benchmarks the MSM with Pippenger with four window-sizes (1, 2, 3 and 4).

It groups the benchmarks so they are shown nicely in the report.